07_errors/patrickmarabeas/mapStore.go

```diff
@@ -9,43 +9,50 @@ func NewMapStore() Storer {
 }
 
 // Create increments the uuid and adds the provided Puppy struct to the store with this identifier.
-func (store *MapStore) Create(puppy Puppy) int {
+func (store *MapStore) Create(puppy Puppy) (int, error) {
+	if puppy.Value < 0 {
+		return -1, NewError(NegativeValue)
```

Comment:
This should be `return 0, NewError(NegativeValue)`

When an error is returned, the value of the other return parameters should be the zero value for their type. Otherwise people may start checking the wrong value for errors, or try to use those values thinking there might be some significance to the non-zero-value.